EIPS/eip-8888.md

```diff
@@ -0,0 +1,195 @@
+
+## Specification
+
+The canonical genesis file MUST be a JSON object with the following top-level fields:
+
+### Top-Level Fields
+
+| Field           | Description                                                     |
+|-----------------|-----------------------------------------------------------------|
+| `config`        | Chain configuration object.                                     |
+| `alloc`         | Map of addresses to pre-allocated balances and/or code/storage. |
+| `nonce`         | Hex-encoded nonce (8 bytes).                                    |
+| `timestamp`     | Hex-encoded UNIX timestamp.                                     |
+| `extraData`     | Arbitrary extra data.                                           |
+| `gasLimit`      | Hex-encoded block gas limit.                                    |
+| `difficulty`    | Hex-encoded block difficulty.                                   |
+| `mixhash`       | Hex-encoded mix hash.                                           |
+| `coinbase`      | Hex-encoded address.                                            |
+
+### `config` Object
+
+The `config` object contains hardfork activation block numbers and fork configurations. Known keys include:
+
+| Field                     | Description                                                                      |
+|---------------------------|----------------------------------------------------------------------------------|
+| `chainId`                 | unique identifier for the blockchain.                                            |
+| `<hardfork(Block\|Time)>` | block height or timestamp to activate the named hardfork.                        |
+| `terminalTotalDifficulty` | difficulty after which to switch from PoW to PoS.                                |
+| `depositContractAddress`  | Ethereum address for the deposit contract                                        |
+| `blobSchedule`            | Map of hardforks and their EIP-4844 DAS [configuration parameters](eip-7840.md). |
+
+### `blobSchedule` Object
+
+| Field                   | Description                                  |
+|-------------------------|----------------------------------------------|
+| `target`                | desired number of blobs to include per block |
+| `max`                   | maximum number of blobs to include per block |
+| `baseFeeUpdateFraction` | input to pricing formula per EIP-4844        |
+
+### `alloc` Object
+
+The `alloc` field defines the initial state at genesis. It maps addresses (as lowercase hex strings) to the following object:
+
+| Field          | Description                                     |
+|----------------|-------------------------------------------------|
+| `balance`      | decimal balance in wei.                         |
+| `code`         | Hex-encoded EVM bytecode.                       |
+| `nonce`        | decimal value.                                  |
```

Comment:
We should limit this field to `uint64`, see https://eips.ethereum.org/EIPS/eip-2681